src/bin/ion/transcribe.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{bail, Result};
+use ion_rs::*;
+use std::io::Write;
+
+/// Constructs the appropriate writer for the given format, then writes all values from the
+/// `Reader` to the new `Writer`.
+pub(crate) fn write_all_as<I: IonInput>(
+    reader: &mut Reader<AnyEncoding, I>,
+    output: &mut impl Write,
+    format: &str,
+) -> Result<usize> {
+    write_n_as(reader, output, format, usize::MAX)
+}
+
+/// Constructs the appropriate writer for the given format, then writes up to `count` values from the
+/// `Reader` to the new `Writer`.
+pub(crate) fn write_n_as<I: IonInput>(
+    reader: &mut Reader<AnyEncoding, I>,
+    output: &mut impl Write,
+    format: &str,
+    count: usize,
+) -> Result<usize> {
+    let written = match format {
+        "pretty" => {
+            let mut writer = Writer::new(v1_0::Text.with_format(TextFormat::Pretty), output)?;
+            transcribe_n(reader, &mut writer, count)
+        }
+        "text" => {
+            let mut writer = Writer::new(v1_0::Text.with_format(TextFormat::Compact), output)?;
+            transcribe_n(reader, &mut writer, count)
+        }
+        "lines" => {
+            let mut writer = Writer::new(v1_0::Text.with_format(TextFormat::Lines), output)?;
+            transcribe_n(reader, &mut writer, count)
+        }
+        "binary" => {
+            let mut writer = Writer::new(v1_0::Binary, output)?;
+            transcribe_n(reader, &mut writer, count)
+        }
+        unrecognized => bail!("unsupported format '{unrecognized}'"),
+    }?;
```

Comment:
Would something like this work?

```suggestion
    let writer_config = match format {
        "pretty" => v1_0::Text.with_format(TextFormat::Pretty),
        "text" => v1_0::Text.with_format(TextFormat::Compact),
        "lines" => v1_0::Text.with_format(TextFormat::Lines),
        "binary" => v1_0::Binary,
        unrecognized => bail!("unsupported format '{unrecognized}'"),
    }?;
    
    let mut writer = Writer::new(writer_config, output)?;
    let written = transcribe_n(reader, &mut writer, count)?;
```

Reply:
The trouble is that the first three branches return `WriteConfig<v1_0::Text>` and the last one returns `WriteConfig<v1_0::Binary>`.